src/systemTest/java/com/flightstats/hub/system/functional/TimedWebhooksTest.java

```diff
@@ -0,0 +1,131 @@
+package com.flightstats.hub.system.functional;
+
+import com.flightstats.hub.model.Webhook;
+import com.flightstats.hub.model.WebhookType;
+import com.flightstats.hub.system.config.DependencyInjector;
+import com.flightstats.hub.system.service.CallbackService;
+import com.flightstats.hub.system.service.ChannelService;
+import com.flightstats.hub.system.service.WebhookService;
+import com.flightstats.hub.utility.StringHelper;
+import lombok.extern.slf4j.Slf4j;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+@Slf4j
+class TimedWebhooksTest extends DependencyInjector {
+    @Inject
+    @Named("test.data")
+    private String testData;
+    private static final int CHANNEL_COUNT = 5;
+    private static final int TIMEOUT = 5 * 60;
+
+    private final List<String> channels = new ArrayList<>();
+    private final List<Webhook> webhooks = new ArrayList<>();
+    private final ConcurrentHashMap<String, ConcurrentLinkedQueue<String>> channelItemsPosted = new ConcurrentHashMap<>();
+    @Inject
+    private StringHelper stringHelper;
+    @Inject
+    private ChannelService channelService;
+    @Inject
+    private WebhookService webhookService;
+    @Inject
+    private CallbackService callbackService;
+
+    @AfterEach
+    void cleanup() {
+        channels.forEach(channelService::delete);
+        webhooks.forEach(webhook -> webhookService.delete(webhook.getName()));
+        channels.clear();
+        webhooks.clear();
+        channelItemsPosted.clear();
+    }
+
+    private boolean channelAndWebhookFactory(WebhookType type) {
+        try {
+            for (int i = 0; i < CHANNEL_COUNT; i++) {
+                String channelName = stringHelper.randomAlphaNumeric(10);
+                String webhookName = stringHelper.randomAlphaNumeric(10);
+                channelService.create(channelName);
+                Webhook webhook = Webhook.builder()
```

Comment:
It's possible that my branch introduces a model builder that takes care of some of these lines (like 62 and 64)...unless I took that out. Hm.